src/honeybadger/payload.cr

```diff
@@ -8,14 +8,38 @@ module Honeybadger
     # The exception to be rendered.
     getter exception : Exception
 
+    getter context : ContextHash
+
+    def set_context(context_hash : ContextHash) : Nil
+      @context = context_hash
+    end
+
     # Subclasses of Payload must set @exception, but will likely need to
-    # take additional parameters for context.
+    # take additional parameters.
     def initialize(@exception : Exception)
+      @context = ContextHash.new
     end
 
-    # Stub implemented to ease the common paradigm of embedding request
-    # details into the payload.
-    def request_json(builder); end
+    # A basic request object contains just a context object.
+    # Override this to embed actual request details.
+    def request_json(builder)
```

Comment:
Does `HttpPayload` also need to add the context object since it overrides this method?

Reply:
Yep, [I added it there as well.](https://github.com/honeybadger-io/honeybadger-crystal/pull/17/files#diff-b1a83269a89956755ae42d605e6a71adee90dbf77a9d2fac0de8e51a9c19fbceR31-R36)